ocaml/idl/datamodel.ml

```diff
@@ -5249,14 +5249,46 @@ let network_detach_for_vm = call
     ~allowed_roles:_R_VM_POWER_ADMIN
     ()
 
+let network_purpose = Enum ("network_purpose", [
+  "nbd", "Network Block Device service using TLS";
+  "insecure_nbd", "Network Block Device service without integrity or confidentiality: NOT RECOMMENDED";
+  "management", "Management";
+  "storage", "Storage";
+  "guest", "Guest virtual machines";
+  "himn", "Host internal management network on which guests will be assigned a private link-local IP address which can be used to talk XenAPI";
+  "unmanaged", "Not managed by xapi";
+])
+(*
+let network_add_purpose = call
+  ~name:"add_purpose"
+  ~doc:"Give a network a new purpose (if not present already)"
+  ~params:[
+    Ref _network, "network", "The network";
```

Comment:
The arguments for add/remove functions are normally called `self` and `value`.

Reply:
I've never liked that, but fine, I'll change it.